src/java/org/apache/cassandra/db/monitoring/MonitoringTask.java

```diff
@@ -328,31 +379,126 @@ protected abstract static class Operation
          * this is set lazily as it takes time to build the query CQL */
         private String name;
 
+        /**
+         * creation time of this Operation object, in ms,
+         * this is different from operation's creationTimeNanos
+         * which does not follow wall clock and is useless for
+         * reporting purposes e.g. in virtual tables
+         */
+        private final long timestamp;
+
+        // optional keyspace and table this operation acts on
+        // used upon deserialisation
+        private String keyspace;
+        private String table;
+        private boolean crossNode;
+
         Operation(Monitorable operation, long failedAtNanos)
         {
             this.operation = operation;
             numTimesReported = 1;
             totalTimeNanos = failedAtNanos - operation.creationTimeNanos();
             minTime = totalTimeNanos;
             maxTime = totalTimeNanos;
+            timestamp = Clock.Global.currentTimeMillis();
```

Comment:
should we calculate the real operation creation timestamp, instead of using a reporting timestamp here (it can be confusing for readers of the table)? We can calculate real timestamp as: currentTimeMillis - (timeNano - creationTimeNanos) / 1_000_000

Reply:
ok, why not